### PR TITLE
Document deprecated leader annotation

### PR DESCRIPTION
**Pull request by kundan2707**
documented unregistered annotation `"control-plane.alpha.kubernetes.io/leader"`
fixes: #39297

### CODE REVIEW

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 1]
```diff
@@ -1086,6 +1086,23 @@ has been truncated to 1000.
...
+
+Used on: Endpoints
+
+The {{< glossary_tooltip text="control plane" term_id="control-plane" >}} adds this annotation to
+an [Endpoints](/docs/concepts/services-networking/service/#endpoints) object. This annotation provides
+following detail.
+- Who is the current leader.
+- The time when the current leadership was acquired:
+- The duration of the lease (of the leadership) in seconds:
+- The time the current lease (the current leadership) should be renewed:
```
Comment (windsonsea) [inline content/en/docs/reference/labels-annotations-taints/_index.md:1103]:
```suggestion
- The time when the current leadership was acquired.
- The duration of the lease (of the leadership) in seconds.
- The time the current lease (the current leadership) should be renewed.
```

**Comment by kundan2707** [comment]
@windsonsea  pushed the changes suggested by you

**Comment by windsonsea** [comment]
lgtm for the text. Could you squash your commits into a single one?

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 2]
```diff
@@ -1086,6 +1086,23 @@ has been truncated to 1000.
 
 If the number of backend endpoints falls below 1000, the control plane removes this annotation.
 
+### control-plane.alpha.kubernetes.io/leader
+
+Type: Annotation
+
+Example: `control-plane.alpha.kubernetes.io/leader={"holderIdentity":"controller-0","leaseDurationSeconds":15,"acquireTime":"2023-01-19T13:12:57Z","renewTime":"2023-01-19T13:13:54Z","leaderTransitions":1}`
```
Comment (sftim) [inline content/en/docs/reference/labels-annotations-taints/_index.md:1093]: I'm afraid this isn't a valid representation of a YAML snippet. Try copying this from an actual Endpoints manifest.

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 3]
```diff
@@ -1086,6 +1086,23 @@ has been truncated to 1000.
 
 If the number of backend endpoints falls below 1000, the control plane removes this annotation.
 
+### control-plane.alpha.kubernetes.io/leader
```
Comment (sftim) [inline content/en/docs/reference/labels-annotations-taints/_index.md:1089]:
```suggestion
### control-plane.alpha.kubernetes.io/leader (deprecated) {#control-plane-alpha-kubernetes-io-leader}
```

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 4]
```diff
@@ -1086,6 +1086,23 @@ has been truncated to 1000.
...
+
+Type: Annotation
+
+Example: `control-plane.alpha.kubernetes.io/leader={"holderIdentity":"controller-0","leaseDurationSeconds":15,"acquireTime":"2023-01-19T13:12:57Z","renewTime":"2023-01-19T13:13:54Z","leaderTransitions":1}`
+
+Used on: Endpoints
+
+The {{< glossary_tooltip text="control plane" term_id="control-plane" >}} adds this annotation to
```
Comment (sftim) [inline content/en/docs/reference/labels-annotations-taints/_index.md:1097]:
```suggestion
The {{< glossary_tooltip text="control plane" term_id="control-plane" >}} previously set annotation on
```

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 5]
```diff
@@ -1086,6 +1086,23 @@ has been truncated to 1000.
...
+an [Endpoints](/docs/concepts/services-networking/service/#endpoints) object. This annotation provides
+following detail.
+- Who is the current leader.
+- The time when the current leadership was acquired.
+- The duration of the lease (of the leadership) in seconds.
+- The time the current lease (the current leadership) should be renewed.
+- The number of leadership transitions that happened in the past.
+
```
Comment (sftim) [inline content/en/docs/reference/labels-annotations-taints/_index.md:1105]:
```suggestion

Kubernetes now uses [Leases](/docs/concepts/architecture/leases/) to
manage leader assignment for the Kubernetes control plane.

```

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 6]
```diff
@@ -1086,6 +1086,23 @@ has been truncated to 1000.
...
+### control-plane.alpha.kubernetes.io/leader
+
+Type: Annotation
+
+Example: `control-plane.alpha.kubernetes.io/leader={"holderIdentity":"controller-0","leaseDurationSeconds":15,"acquireTime":"2023-01-19T13:12:57Z","renewTime":"2023-01-19T13:13:54Z","leaderTransitions":1}`
+
+Used on: Endpoints
+
+The {{< glossary_tooltip text="control plane" term_id="control-plane" >}} adds this annotation to
+an [Endpoints](/docs/concepts/services-networking/service/#endpoints) object. This annotation provides
```
Comment (sftim) [inline content/en/docs/reference/labels-annotations-taints/_index.md:1098]:
```suggestion
an [Endpoints](/docs/concepts/services-networking/service/#endpoints) object. This annotation provided
```

**Review by sftim — COMMENTED** [review]
Thanks

/retitle Document deprecated leader annotation

**Comment by kundan2707** [comment]
> lgtm for the text. Could you squash your commits into a single one?

@windsonsea  i will squash all commits after resolving @sftim  comments.

File: content/en/docs/reference/labels-annotations-taints/_index.md (hunk 2 above)
Comment (kundan2707) [reply content/en/docs/reference/labels-annotations-taints/_index.md:1093]: @sftim  this is not yaml snippet rather it is outcome of kubectl get command corresponding to this annotation

Comment (sftim) [reply content/en/docs/reference/labels-annotations-taints/_index.md:1093]: Thing is: all the other labels and annotations have their examples in the format of a YAML snippet. Let's follow that convention.

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 7]
```diff
@@ -1086,6 +1086,26 @@ has been truncated to 1000.
...
+
+Type: Annotation
+
+Example: `control-plane.alpha.kubernetes.io/leader={"holderIdentity":"controller-0","leaseDurationSeconds":15,"acquireTime":"2023-01-19T13:12:57Z","renewTime":"2023-01-19T13:13:54Z","leaderTransitions":1}`
+
+Used on: Endpoints
+
+The {{< glossary_tooltip text="control plane" term_id="control-plane" >}} previously set annotation on
+an [Endpoints](/docs/concepts/services-networking/service/#endpoints) object. This annotation provided
+following detail.
```
Comment (sftim) [inline content/en/docs/reference/labels-annotations-taints/_index.md:1099]: (nit)
```suggestion
the following detail:

```

File: content/en/docs/reference/labels-annotations-taints/_index.md (hunk 2 above)
Comment (kundan2707) [reply content/en/docs/reference/labels-annotations-taints/_index.md:1093]: @sftim  as its got deprecated i am not able to find exact example in any yaml file. 
Can you help me in this?

Comment (kundan2707) [reply content/en/docs/reference/labels-annotations-taints/_index.md:1093]: @sftim  as its got deprecated i am not able to find exact example in any yaml file. 
Can you help me in this?

Comment (sftim) [reply content/en/docs/reference/labels-annotations-taints/_index.md:1093]: You don't need to copy this from an actual Endpoints; another way is to make sure you are writing [a fragment of] valid YAML.

Try this:
https://learnxinyminutes.com/docs/yaml/

You've got the string value right, but you haven't quoted it, and you've used an `=` where YAML uses a `:`.

**Comment by tengqm** [comment]
@kundan2707 Please check the latest comment and address it if possible.
/approve

**Comment by kundan2707** [comment]
@tengqm  will incorporate latest comment

File: content/en/docs/reference/labels-annotations-taints/_index.md (hunk 6 above)
Comment (tengqm) [reply content/en/docs/reference/labels-annotations-taints/_index.md]: @kundan2707 In your next PR, please help mark these conversations as "resolved" by clicking the "Resolve conversation" button below when you agree to the suggests and you have revised the PR accordingly.
Checking conversation resolved helps the reviewers confirm that previous discussions are clean, the current reviewer may switch to different aspects of the PR, and a new reviewer wouldn't need to check the whole history of a PR.

**Comment by tengqm** [comment]
All feedback resolved.
/lgtm
(reapplied).

**Pull request merged**